devito/parameters.py

```diff
@@ -94,8 +94,10 @@ def name(self):
         return self._name
 
     def _signature_items(self):
-        items = sorted(it for it in self.items()
-                       if it[0] not in ['log_level', 'first_touch'])
+        # Note: we are discarding some vars that do not affect the c level
+        # code in order to avoid recompiling when such vars are modified
+        discard = ['profiling', 'autotuning', 'log_level', 'first_touch']
```

Comment:
now that I think about it, 'profiling' actually affects the C code as Advisor does something different than basic/advanced

Reply:
Hmm, ok. I'll look at this in more detail tomorrow then. Thanks.